File: Contract.Architecture.Frontends/Contract.Architecture.Web.Core/src/app/model/bankwesen/banken/dtos/i-bank.ts
import { ApiBank } from './api/api-bank';

export interface IBank {
    id: string;
    name: string;
    eroeffnetAm: Date;
    isPleite: boolean;
}

export class Bank {
    public static fromApiBank(apiBank: ApiBank): IBank {
        return {
            id: apiBank.id,
            name: apiBank.name,
            eroeffnetAm: apiBank.eroeffnetAm,
            isPleite: apiBank.isPleite,
        };
    }
}

File: Contract.Architecture.Backends/Contract.Architecture.Backend.Core/Contract/Logic/Modules/Bankwesen/Banken/DTOs/IBank.cs
using System;

namespace Contract.Architecture.Backend.Core.Contract.Logic.Modules.Bankwesen.Banken
{
    public interface IBank
    {
        Guid Id { get; set; }

        string Name { get; set; }

        DateTime EroeffnetAm { get; set; }

        bool IsPleite { get; set; }
    }
}